android/app/src/main/java/com/zulipmobile/TextCompressionModule.java

```diff
@@ -0,0 +1,98 @@
+package com.zulipmobile;
+
+import android.util.Base64;
+
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.Promise;
+
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+import android.util.Log;
+
+class TextCompressionModule extends ReactContextBaseJavaModule {
+  public TextCompressionModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "TextCompressionModule";
+  }
+
+  // TODO: Experiment what value gives the best performance.
+  private final int bufferSize = 8192;
+
+  private final String header = "z|base64 zlib|";
+
+  @Override
+  public Map<String, Object> getConstants() {
```

Comment:
The nice thing about getConstants is that React uses it to store the constants directy in JS and thus avoids making a roundtrip over the bridge when requesting the constants.